app/models/katello/erratum.rb

```diff
@@ -52,13 +52,11 @@ def self.applicable_to_hosts(hosts)
       # which is calculated elsewhere.
 
       self.joins(:content_facets).
-        where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts)
+        where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts.to_a)
```

Comment:
this won't perform well with large numbers of hosts.  I'll see if i can figure out why its failing

Reply:
`to_sql` works as well, without the performance hit, but some tests fail that try to pass an array instead of an `ActiveRecord::Association`, so they'd need to be modified.

Reply:
Oh, I think `hosts.pluck(:id)` would also work without the performance hit. Again, arrays passed in for tests would fail.

Reply:
.select(:id) would be even more ideal if that works, if now lets see if we can figure out why these things aren't working

Reply:
Aha! `.select(:id)` does exhibit the same behavior as the prior code.

```
Hirb Error: PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "compute_resources"
LINE 1: ..." WHERE "hosts"."type" IN ('Host::Managed') AND (("compute_r..
```

Reply:
Playing with this it seems that we are hitting a bug in either Active Record or scoped search (or their interaction). 

I think if we're not going to try to identify the underlying bug, changing it to .pluck(:id) would be the best thing to do.  .to_a would cause the full record to be fetched rather than just the id.